clippy_lints/src/methods/option_map_or_err_ok.rs

```diff
@@ -0,0 +1,45 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet;
+use clippy_utils::ty::is_type_diagnostic_item;
+use clippy_utils::{is_res_lang_ctor, path_res};
+use rustc_errors::Applicability;
+use rustc_hir::LangItem::{ResultErr, ResultOk};
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_span::symbol::sym;
+
+use super::OPTION_MAP_OR_ERR_OK;
+
+pub(super) fn check<'tcx>(
+    cx: &LateContext<'tcx>,
+    expr: &'tcx Expr<'tcx>,
+    recv: &'tcx Expr<'_>,
+    or_expr: &'tcx Expr<'_>,
+    map_expr: &'tcx Expr<'_>,
+) {
+    let is_option = is_type_diagnostic_item(cx, cx.typeck_results().expr_ty(recv), sym::Option);
+
+    if !is_option {
+        return;
+    }
+
+    let map_arg_is_ok = is_res_lang_ctor(cx, path_res(cx, map_expr), ResultOk);
+
+    if map_arg_is_ok
```

Comment:
I think those could all be pulled into the same `if` condition, no need to define variables.

I guess this would also apply to your previous PR about another of those lints, but no need to go back to that in this PR.

Reply:
Ok, I'll send a follow-up. I took this approach because I thought it made code easier to read. I'll just add comments instead.